frontend/src/components/pages/FridgeManagement/FridgeCheckIns/index.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Button,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  HStack,
+  Input,
+  Text,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { isAfter, parse } from "date-fns";
+import React, { useState } from "react";
+import DatePicker, { DateObject } from "react-multi-date-picker";
+import { useHistory } from "react-router-dom";
+import CheckInAPIClient from "../../../../APIClients/CheckInAPIClient";
+import * as Routes from "../../../../constants/Routes";
+import { CreateCheckInFields } from "../../../../types/CheckInTypes";
+import ErrorMessages from "./ErrorMessages";
+
+const CreateCheckIn = () => {
+  const toast = useToast();
+  const history = useHistory();
+  const [dateRange, setDateRange] = useState<DateObject[]>([
+    new DateObject(),
+    new DateObject().add(1, "days"),
+  ]);
+  const [startTime, setStartTime] = useState<string>();
+  const [endTime, setEndTime] = useState<string>();
+  const [notes, setNotes] = useState<string>();
+  const [formErrors, setFormErrors] = useState({
+    timeRange: "",
+    dateRange: "",
+  });
+
+  const validateForm = () => {
+    const newErrors = {
+      timeRange: "",
+      dateRange: "",
+    };
+    let valid = true;
+    if (!startTime || !endTime) {
+      valid = false;
+      newErrors.timeRange = ErrorMessages.bothTimeFieldsRequired;
+    } else if (
+      isAfter(
+        parse(startTime!, "kk:mm", new Date()),
+        parse(endTime!, "kk:mm", new Date()),
+      )
+    ) {
+      valid = false;
+      newErrors.timeRange = ErrorMessages.endTimeBeforeStartTime;
+    }
+    if (!dateRange[0] || !dateRange[1]) {
+      valid = false;
+      newErrors.dateRange = ErrorMessages.bothDateFieldsRequired;
+    }
+    setFormErrors(newErrors);
+    return valid;
+  };
+
+  const onSaveClick = async () => {
+    if (!validateForm()) {
+      return;
+    }
+
+    const checkInData: CreateCheckInFields = {
+      startDate: parse(
+        startTime!,
```

Comment:
also here and below, isn't it better instead of using a postfix to assert non-null, that we just set an initial default `startTime` and `endTime`?